### PR TITLE
fix: Fix feature view __getitem__ for feature services

### LINKED ISSUE #2767: Feature services do not use specified features correctly
## Expected Behavior 

Given a minimal example:

```
file_source = prep_file_source(df=simple_dataset_1, timestamp_field="ts_1") as file_source:
entity1 = Entity(name="test1", join_keys=["id_join_key"])
feature_view_1 = FeatureView(
    name="test1", entities=[entity1], source=file_source,
)
feature_service = FeatureService(name="fs_1", features=[feature_view_1[["string_col"]]])
```

The feature service should only contain a single feature, `string_col`. 

## Current Behavior
The feature service should contains all the features for the feature view.

## Steps to reproduce

`test_update_feature_services_with_inferred_features` exhibits this behaviour 

Thanks to the bugreport: https://tectonfeast.slack.com/archives/C01MSKCMB37/p1653998863644589

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2767

### CODE REVIEW

**Comment by adchia** [comment]
looks like unit tests are failing btw!

**Comment by achals** [comment]
> looks like unit tests are failing btw!

Thanks fixed!

File: sdk/python/feast/feature_view_projection.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class FeatureViewProjection:
 
     name: str
     name_alias: Optional[str]
+    desired_features: List[str]
```
Comment (kevjumba) [inline sdk/python/feast/feature_view_projection.py:30]: Is this also a bug for the go feature server?

Comment (achals) [reply sdk/python/feast/feature_view_projection.py:30]: Sorry, what do you mean?

Comment (kevjumba) [reply sdk/python/feast/feature_view_projection.py:30]: The go feature server also implements feature view projection and featureservice. Could this bug also be a bug for the go feature service retrieval?

**Pull request merged**